presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZValueRange.java

```diff
@@ -13,35 +13,48 @@
  */
 package com.facebook.presto.hive.zorder;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
 /**
- * The ZValueRange class contains Optional Integer minimum and maximum values. If existing, they are both inclusive.
+ * The ZValueRange class contains Optional Integer minimum and maximum value lists. If existing, they are all inclusive.
  * <p/>
  * Examples of converting queries into ZValueRange variables:
- * (1) "column == 5"                    -> <code> new ZValueRange(Optional.of(5), Optional.of(5)); </code>
- * (2) "column >= 0 and column <= 9"    -> <code> new ZValueRange(Optional.of(0), Optional.of(9)); </code>
- * (3) "column >= 1"                    -> <code> new ZValueRange(Optional.of(1), Optional.empty()); </code>
- * (4) no range for the column          -> <code> new ZValueRange(Optional.empty(), Optional.empty()); </code>
+ * (1) "column == 5"                    -> <code> new ZValueRange(ImmutableList.of(Optional.of(5)), ImmutableList.of(Optional.of(5))); </code>
+ * (2) "column >= 0 AND column <= 9"    -> <code> new ZValueRange(ImmutableList.of(Optional.of(0)), ImmutableList.of(Optional.of(9))); </code>
+ * (3) "column >= 1"                    -> <code> new ZValueRange(ImmutableList.of(Optional.of(1)), ImmutableList.of(Optional.empty())); </code>
+ * (4) no range for column              -> <code> new ZValueRange(ImmutableList.of(Optional.empty()), ImmutableList.of(Optional.empty())); </code>
+ * (5) "column <= 5 OR column >= 10"    -> <code> new ZValueRange(ImmutableList.of(Optional.empty(), Optional.of(10)), ImmutableList.of(Optional.of(5), Optional.empty())); </code>
  */
 public class ZValueRange
 {
-    private final Integer minimumValue;
-    private final Integer maximumValue;
+    private final List<Integer> minimumValues;
+    private final List<Integer> maximumValues;
+    public final int length;
```

Comment:
Create a public function to expose length.
Meanwhile, if I understand correctly each dimension can have multiple intervals in any searches, which is represented by the number of min/max pairs. Therefore, the "length" here means the number of intervals for one dimension. Do you think numOfIntervals or IntervalNum will be better?